hypershift-operator/controllers/hostedcluster/internal/platform/kubevirt/kubevirt.go

```diff
@@ -0,0 +1,209 @@
+package kubevirt
+
+import (
+	"context"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/upsert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sutilspointer "k8s.io/utils/pointer"
+	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1beta1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	hostedClusterAnnotation = "hypershift.openshift.io/cluster"
+	// TODO nargaman replace with pre-built valid image
+	imageCAPK = "registry:5000/capk:devel"
+)
+
+type Kubevirt struct{}
+
+func (p Kubevirt) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
+	hcluster *hyperv1.HostedCluster,
+	controlPlaneNamespace string, apiEndpoint hyperv1.APIEndpoint) (client.Object, error) {
+	kubevirtCluster := &capikubevirt.KubevirtCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      hcluster.Spec.InfraID,
+		},
+	}
+	_, err := controllerutil.CreateOrPatch(ctx, c, kubevirtCluster, func() error {
+		return reconcileKubevirtCluster(kubevirtCluster, hcluster)
+	})
+	if err != nil {
+		return nil, err
+	}
+	kubevirtCluster.TypeMeta = metav1.TypeMeta{
+		Kind:       "KubevirtCluster",
+		APIVersion: capikubevirt.GroupVersion.String(),
+	}
+
+	return kubevirtCluster, nil
+}
+
+func reconcileKubevirtCluster(kubevirtCluster *capikubevirt.KubevirtCluster, hcluster *hyperv1.HostedCluster) error {
+	// We only create this resource once and then let CAPI own it
+	kubevirtCluster.Annotations = map[string]string{
+		hostedClusterAnnotation:    client.ObjectKeyFromObject(hcluster).String(),
+		capiv1.ManagedByAnnotation: "external",
+	}
+	// Set the values for upper level controller
+	kubevirtCluster.Status.Ready = true
+	return nil
+}
+
+func (p Kubevirt) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, tokenMinterImage string) (*appsv1.DeploymentSpec, error) {
+	defaultMode := int32(420)
+	capkLabels := map[string]string{
```

Comment:
Let's drop this lables as we don't really need this labels, the generic func sets them.

Reply:
Done